src/eos-crash-metrics.c

```diff
@@ -120,21 +123,40 @@ get_ostree_repo_url (OstreeRepo *repo, const char *origin)
   return url;
 }
 
-static char *
-get_eos_ostree_deployment_commit (OstreeSysroot *sysroot, OstreeRepo *repo)
+static gboolean
+get_eos_ostree_deployment_commit (OstreeSysroot *sysroot,
+                                  OstreeRepo    *repo,
+                                  char         **commit_out,
+                                  char         **version_out)
 {
   OstreeDeployment *deployment = ostree_sysroot_get_booted_deployment (sysroot);
   g_autoptr(GError) error = NULL;
   const char *csum = NULL;
+  g_autoptr(GVariant) commit = NULL;
+  g_autoptr(GVariant) commit_metadata = NULL;
+  const char *version = NULL;
 
   if (!deployment)
     {
       g_warning ("OSTree deployment is not currently booted, cannot read state");
-      return NULL;
+      return FALSE;
     }
 
   csum = ostree_deployment_get_csum (deployment);
-  return g_strdup (csum);
+
+  /* Load the backing commit; shouldn't normally fail, but if it does,
+   * we stumble on.
+   */
+  if (ostree_repo_load_variant (repo, OSTREE_OBJECT_TYPE_COMMIT, csum,
+                                &commit, NULL))
+    commit_metadata = g_variant_get_child_value (commit, 0);
+
+  if (commit_metadata)
+    (void) g_variant_lookup (commit_metadata, OSTREE_COMMIT_META_KEY_VERSION, "&s", &version);
+
+  *commit_out = g_strdup (csum);
+  *version_out = g_strdup (version);
```

Comment:
Nitpick: Would be good to either `g_return_val_if_fail (commit_out != NULL, FALSE)` for each of these, or `if (commit_out != NULL) *commit_out = g_strdup (…)`

Reply:
Addressed by adding a `g_return_val_if_fail` guard.